presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/security/FileBasedAccessControl.java

```diff
@@ -211,10 +212,28 @@ public void checkCanShowColumns(ConnectorSecurityContext context, SchemaTableNam
     @Override
     public List<ColumnMetadata> filterColumns(ConnectorSecurityContext context, SchemaTableName tableName, List<ColumnMetadata> columns)
     {
-        if (!checkAnyTablePermission(context, tableName)) {
+        if (INFORMATION_SCHEMA_NAME.equals(tableName.getSchemaName())) {
+            return columns;
+        }
+
+        ConnectorIdentity identity = context.getIdentity();
+        TableAccessControlRule rule = tableRules.stream()
+                .filter(tableRule -> tableRule.matches(identity.getUser(), identity.getGroups(), tableName))
+                .findFirst()
+                .orElse(null);
```

Comment:
Maybe you could use some predefined rule instead of using `null`. The `if` below would be easier. ?